### PR TITLE
Remove extra blank lines in Protobuf generated code

### LINKED ISSUE #3759: Extra blank line in ServiceTests.IceRpc.cs
This generated file is currently the only Protobuf service tests and shows some bugs:

```csharp
public partial interface IMyWidget
{
     <-- extra blank line   
}

...

 }

  <-- extra blank lime   
}

```

### PR DESCRIPTION
**Pull request by pepone**
Fix #3759

I added a helper method to remove that allows removing extra empty lines, because I think is nice to be able to use the string templates in the generators. I think the string templates make it easy to see what the generator produces.

### CODE REVIEW

**Comment by externl** [comment]
Did you consider writing a builder like we do for the rust code generator? This will probably be useful in the for a future C# version of the Slice code generator.

**Comment by pepone** [comment]
>  We can't just not generate these empty lines to start with?

> I don't see how this relates to our use of string templates.

```
 return @$"
/// <remarks>protoc-gen-icerpc-csharp generated this client-side interface from Protobuf service <c>{service.FullName}</c>.
/// It's implemented by <c>{service.Name.ToPascalCase()}Client</c></remarks>
public partial interface I{service.Name.ToPascalCase()}
{{
    {methods.Trim()}
}}";
```

Consider the above template: if `methods` is empty, it will result in an empty line. We could structure the code differently, using if/else statements to avoid adding empty lines in the first place. However, I find that the current approach simplifies understanding what is being generated

> Did you consider writing a builder like we do for the rust C# code generator?

This started as a simple protoc plug-in, and using plain strings was the simpler way to get things rolling.

**Review by externl — APPROVED** [review]
Looks good. I think it could be nice to replace this with a builder pattern in the future.

File: tools/IceRpc.ProtocGen/StringExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+// Copyright (c) ZeroC, Inc.
+
+namespace IceRpc.ProtocGen;
+
+internal static class StringExtensions
+{
+    /// <summary>Removes consecutive empty lines, and empty lines that are between an open and a close brace '{', '}'
```
Comment (bernardnormier) [inline tools/IceRpc.ProtocGen/StringExtensions.cs:7]: As far as I can tell, the only superfluous empty lines we currently generate are:
```
 -- extra empty line here
}
```

And an empty line before a closing } is always incorrect. So can we simplify this code?

Comment (pepone) [reply tools/IceRpc.ProtocGen/StringExtensions.cs]: You are right, I simplified the code.

**Pull request merged**